### ISSUE TITLE
Machine classification succeeds, but fails to mark features as classified

### ISSUE THREAD

**Issue by StephenChan** [beijbom/coralnet#447]
This has been happening the last few days (at least as early as May 9), and has led to the following observed issues:

- https://groups.google.com/g/coralnet-users/c/LY4uP86GcjE
- https://groups.google.com/g/coralnet-users/c/alpUF0sqnn8 (Alleviate runs into a `list index out of range` error)

Don't know what the cause could be yet.

**Comment by StephenChan** [comment]
Two types of errors that I'm seeing repeated in the celeryworker logs:

```
[2022-05-13 12:11:46,134: ERROR/MainProcess] Task Classify Image[e269ac92-a5ef-4e5e-9800-b469460c7f2c] raised unexpected: KeyError((279, 1032),)
  File "... site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "... site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "... coralnet/project/vision_backend/tasks.py", line 264, in classify_image
    th.add_annotations(image_id, res, label_objs, classifier)
  File "... site-packages/reversion/revisions.py", line 324, in do_revision_context
    return func(*args, **kwargs)
  File "... coralnet/project/vision_backend/task_helpers.py", line 79, in add_annotations
    scores = res[(point.row, point.column)]
  File "... site-packages/spacer/messages.py", line 340, in __getitem__
    return rc_set[rowcol]
KeyError: (279, 1032)
```

```
[2022-05-13 14:37:48,482: ERROR/MainProcess] Task Classify Image[2651adea-6cff-472b-b8ca-f5e290bc9238] raised unexpected: DataError('integer out of range\n',)
Traceback (most recent call last):
  File "... site-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "... site-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "... coralnet/project/vision_backend/tasks.py", line 277, in classify_image
    th.add_scores(image_id, res, label_objs)
  File "... coralnet/project/vision_backend/task_helpers.py", line 129, in add_scores
    Score.objects.bulk_create(score_objs)
  File "... site-packages/django/db/models/manager.py", line 82, in manager_method
[truncated]
  File "... site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "... site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.DataError: integer out of range
```

The first such error I'm seeing in the logs is: `[2022-05-09 02:02:22,460: ERROR/MainProcess] Task Classify Image[8caca78f-e74c-4422-a85f-c3c67ba2673c] raised unexpected: KeyError((754, 1025),)` (that's Pacific time).

**Comment by StephenChan** [comment]
Huh, well that's unexpected. I think the `Score` table IDs have actually reached the max 32-bit signed integer value. `Score.objects.latest('pk').pk` gives 2147381607 as the latest non-deleted Score ID in the database. `2**31 - 1` is 2147483647.

Looks like we'll have to migrate Score's ID column to a bigger integer type: [BigAutoField](https://docs.djangoproject.com/en/dev/ref/models/fields/#bigautofield) instead of [AutoField](https://docs.djangoproject.com/en/dev/ref/models/fields/#autofield).

Other big tables for comparison:

- django-reversion's Version is at 267743341 (about 8 times less)
- Annotation is at 177933251 (about 12 times less)
- Point is at 133602130 (about 16 times less)

**Comment by StephenChan** [comment]
PR #450 ported Score's id field to a BigAutoField.

This was just an in-place column alteration. To estimate the required time, I went onto the staging server and tried a similar migration on Annotations (the largest table in the staging server's database, 10 million rows). That took 200 seconds to run. So I multiplied that by 215 (according to the number of Score rows in the production database) to get an estimate of 12 hours. It actually finished in 1 hour, though!

If the estimate ended up being multiple days, I may have considered a more complex migration which would only need downtime for a small portion of the migration (like in [this article](https://tech.coffeemeetsbagel.com/reaching-the-max-limit-for-ids-in-postgres-6d6fa2b1c6ea)). But I figured 12 hours on a weekend was OK for our standards, so I went the easier way.

After that migration finished, I went to the Backend Overview page, and identified the recently affected sources as the red-highlighted rows that have unclassified > 0 and verified robots > 0. I ran the following on those sources to classify their "unclassified" images (which includes actually unclassified images, and images which failed to get scores saved):

```
from images.models import Source, Image
from vision_backend.tasks import classify_image
for source_id in [<SOURCE IDS GO HERE>]:
    source = Source.objects.get(pk=source_id)
    images = source.image_set.filter(annoinfo__confirmed=False, features__classified=False)
    for image in images:
        classify_image.delay(image.pk)
```

I'm not seeing DataError in the logs now, and the latest Score ID is now 2152444372 (greater than 2147483647). And the Backend Overview queue has diminished as expected. So I think we're good here.

(Incidentally, Backend Overview took like 40 minutes to load once, so I'm seeing some awful performance at times - but that's another issue.)
